### PR TITLE
Add fill selfs food info --> fillAllSelfFoodInfo

### PR DESCRIPTION
**Pull request by hosseindehghanipour1998**
… MealInfo , fillAvailableMeals() -> MealInfo

### CODE REVIEW

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java [hunk 1]
```diff
@@ -243,22 +243,28 @@ public Student studentLogin(String userName, String password) {
...
             //Fill MealInfo
             fillStudentMealInfo(newStudent, list.get("coupons"));
-            sendPostRequest(tokenToJson(userToken),logoutURL);
+
 
             //Fill Self FoodInfo
             fillSelfFoodInfo(list.get("self_data"));
 
+
+            //Logging out
```
Comment (aminsoheyli) [inline app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java]: 👍

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MealInfo.java [hunk 2]
```diff
@@ -10,10 +14,11 @@
     private int reservedFoodId;
     private Self reservedSelf;
     private MealType mealType;
+    private String key_id ;
```
Comment (aminsoheyli) [inline app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MealInfo.java]: What is it exactly?

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java [hunk 3]
```diff
@@ -24,10 +30,11 @@
     private static String loginURL = "http://nowaw.pythonanywhere.com/login/login";
     private static String studentInfoURL = "http://nowaw.pythonanywhere.com/login/self_data";
     private static String logoutURL = "http://nowaw.pythonanywhere.com/login/logout";
+    private static String deleteURL = "http://nowaw.pythonanywhere.com/login/delete";
     private static HttpURLConnection connection;
     private static String USER_AGENT = "Mozilla/5.0";
     private static String token;
```
Comment (aminsoheyli) [inline app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:36]: @hosseindehghanipour1998 
Is this token different from Student.user_token?

**Review by Maaz77 — COMMENTED** [review]
MealInfo.allAvailableMealInfo this array filled successfully but all couponID of the members of this array are zero and i have no idea about key_id.

**Comment by hosseindehghanipour1998** [comment]
@Maaz77 Happy to hear that the functions have worked properly ... i saw that in the JSON given by server we had a key_id not a coupon_id for self foods. i would remove it if you wish and replace the data in coupon_id. There is no distinction between them ( they are both the same ) and i will fix it now. Assume that there is no key_id and use coupon_id instead.

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java (hunk 3 above)
Comment (hosseindehghanipour1998) [reply app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:36]: Nope they are both the same but I couldn't reach the token sent by the server in IONIC function so 
 I had to define a static token. I will inform you later on about this specific variable.

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java [hunk 4]
```diff
@@ -245,19 +243,28 @@ public Student studentLogin(String userName, String password) {
...
             //Fill MealInfo
             fillStudentMealInfo(newStudent, list.get("coupons"));
+
+
+            //Fill Self FoodInfo
+            fillSelfFoodInfo(list.get("self_data"));
+
+
+            //Logging out
             sendPostRequest(tokenToJson(userToken),logoutURL);
```
Comment (hosseindehghanipour1998) [inline app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:264]: @aminsoheyli this is the log out function that maaz has used to log out.

Comment (hosseindehghanipour1998) [reply app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:264]: @aminsoheyli did you delete that protected function ?
I suggest not to delete that untill i find out wether it has any negative impact on the project or not.

Comment (aminsoheyli) [reply app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:264]: Yeah, I seen it.

**Comment by Maaz77** [comment]
@hosseindehghanipour1998 clean up your whole code from any extra comment line and extra variable and also Log.d then merge and extra functions

File: app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java (hunk 4 above)
Comment (aminsoheyli) [reply app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:264]: > @aminsoheyli did you delete that protected function ?
> I suggest not to delete that untill i find out wether it has any negative impact on the project or not.

@hosseindehghanipour1998 No, I don't manipulate this code until your last commit.

Comment (aminsoheyli) [reply app/src/main/java/ir/ac/shirazu/softwareproject/server_api/Meal/MyKit.java:264]: @hosseindehghanipour1998  Has it been completed?

**Pull request merged**